### ISSUE TITLE
Suggest `rustup update` before getting started

### ISSUE THREAD

**Issue by kornelski** [rust-lang/www.rust-lang.org#699]
# Content Feature Request
## Summary

Suggest updating Rust in case user has an old version that won't work with the examples.

Users who have installed Rust in the past, but only now are getting around to trying it (again), have pre-2018 version, and for them example code seems broken, e.g.:

https://twitter.com/willkirkby/status/1090757489518432261

## Motivation
Rust is unusual with its short "expiration date" of the compiler, so users may assume that if they already have (some) version of Rust, it should be enough at least for a hello world project.

## Drawbacks
It's an extra step in the tutorial.

## Rationale and alternatives
- Rust's ecosystem is quick adopting latest features, so having less-than-newest compiler is likely to give an impression that lots of crates are broken. It may be useful to be upfront about this.
- 2015/2018 jump is bigger than others, as it affects even basic tutorial. Doing nothing is an option, because over time number of people with outdated installs will naturally dwindle, but IMHO it's worth addressing this issue now.
- In this case it's too late for https://github.com/rust-lang/rfcs/pull/2495, but that RFC could help in similar cases in the future.

## Maintenance
I can make a PR to add a paragraph about `rustup update`, and perhaps remove it in, say a year, if it turns out to be a transient problem.

**Comment by skade** [comment]
@kornelski Which page does this exactly apply to? I'm happy to review and merge a PR.

**Comment by kornelski** [comment]
The relevant page is https://www.rust-lang.org/learn/get-started - the ferris says example is the code that doesn't work with 2015 rust. I must admit I'm not sure exactly where exactly to put the rustup update suggestion, as the process is split over several sections.

**Comment by zanedp** [comment]
I don't know what version of `cargo` I had, but `cargo new hello-rust` made a new library instead of a binary, meaning all the rest of the doc was basically wrong. When I stumbled across this "issue", I used `rustup update`, tried again, and got matching output. So, the problem extends even beyond crate incompatibilities. (I had just run `rustup-init.exe`, and it was not clear that wasn't installing the latest version of everything for me.)
